src/main/java/de/retest/web/AttributesProvider.java

```diff
@@ -27,32 +35,29 @@ public static AttributesProvider getInstance() {
 		return instance;
 	}
 
-	private final AttributesConfig attributesConfig;
-
-	private AttributesProvider() {
-		attributesConfig = readAttributesConfig();
-	}
-
 	private AttributesConfig readAttributesConfig() {
 		final String userAttributesFilePath = System.getProperty( ATTRIBUTES_FILE_PROPERTY );
 		if ( userAttributesFilePath != null ) {
-			final File userAttributes = new File( userAttributesFilePath );
-			logger.debug( "Loading user-defined attribues file '{}'.", userAttributes );
-			return readAttributesConfigFromFile( userAttributes );
+			final Path userAttributes = Paths.get( userAttributesFilePath );
+			logger.debug( "Loading user-defined attributes file '{}'.", userAttributes );
+			try ( final InputStream in = Files.newInputStream( userAttributes ) ) {
+				return readAttributesConfigFromFile( in );
+			} catch ( final IOException e ) {
+				throw new RuntimeException( "Cannot read attributes file '" + userAttributes + "'.", e );
+			}
 		} else {
-			final File defaultAttributes = new File( getClass().getResource( DEFAULT_ATTRIBUTES_FILE_PATH ).getPath() );
-			logger.debug( "Loading default attributes file '{}'", defaultAttributes );
-			return readAttributesConfigFromFile( defaultAttributes );
+			logger.debug( "Loading default attributes file '{}'", DEFAULT_ATTRIBUTES_FILE_PATH );
+			try ( final InputStream url = getClass().getResourceAsStream( DEFAULT_ATTRIBUTES_FILE_PATH ) ) {
+				return readAttributesConfigFromFile( url );
+			} catch ( final IOException e ) {
+				throw new RuntimeException( "Cannot read attributes file '" + DEFAULT_ATTRIBUTES_FILE_PATH + "'.", e );
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define and throw a dedicated exception instead of using a generic one. ([squid:S00112](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS00112))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=37&issues=AWTwNt2zJ-NkmwErzCSB&open=AWTwNt2zJ-NkmwErzCSB)

[ISSUE_KEY]: AWTwNt2zJ-NkmwErzCSB